### PR TITLE
541 patterns support

### PR DESCRIPTION
**Pull request by sankhesh**
Fixes issues #541 #1042

### CODE REVIEW

**Comment by aashish24** [comment]
Great stuff @sankhesh can you get rid of "WIP" from commit messages?

File: Packages/vcs/Lib/fillarea.py [hunk 1]
```diff
@@ -336,6 +355,7 @@ def __init__(self, Tf_name=None, Tf_name_src='default'):
             self._style = ['solid', ]
             self._index = [1, ]
             self._color = [1, ]
+            self._opacity = []
```
Comment (doutriaux1) [inline Packages/vcs/Lib/fillarea.py]: shouldn't we default to 1. ?

File: Packages/vcs/Lib/isofill.py [hunk 2]
```diff
@@ -595,8 +612,9 @@ def __init__(self, Gfi_name, Gfi_name_src='default'):
             self._ext_1 = False
             self._ext_2 = False
             self._fillareastyle = 'solid'
-            self._fillareaindices = None
+            self._fillareaindices = [1, ]
             self._fillareacolors = [1, ]
+            self._fillareaopacity = []
```
Comment (doutriaux1) [inline Packages/vcs/Lib/isofill.py]: same here. shouldn't we default to 1?

File: Packages/vcs/Lib/vcsvtk/boxfillpipeline.py [hunk 3]
```diff
@@ -329,6 +343,7 @@ def _plotInternalCustomBoxfill(self):
                     I = [indices[i]]
         tmpLevels.append(L)
         tmpColors.append(C)
+        print tmpLevels
```
Comment (doutriaux1) [inline Packages/vcs/Lib/vcsvtk/boxfillpipeline.py]: left over?

**Comment by doutriaux1** [comment]
@sankhesh I'm excited. Going to play with it right way! quick question does iso.script("iso.json") work? I don't see you updated the test_dump_json and it should probably fail now since we have a new attribute.

**Comment by doutriaux1** [comment]
I will let @aashish24 do the "commits police" work :stuck_out_tongue_winking_eye:

**Comment by doutriaux1** [comment]
@sankhesh also could you please rebase on master?

File: Packages/vcs/Lib/fillarea.py (hunk 1 above)
Comment (sankhesh) [inline Packages/vcs/Lib/fillarea.py]: Nope. The default is different in different cases.
For solids and patterns, default opacity value is 0 and for hatches it is 100.

File: Packages/vcs/Lib/isofill.py (hunk 2 above)
Comment (sankhesh) [inline Packages/vcs/Lib/isofill.py]: Nope. The default is different in different cases.
For solids and patterns, default opacity value is 0 and for hatches it is 100.

File: Packages/vcs/Lib/vcsvtk/boxfillpipeline.py (hunk 3 above)
Comment (sankhesh) [inline Packages/vcs/Lib/vcsvtk/boxfillpipeline.py]: Good spot. I'll fix it.

**Comment by sankhesh** [comment]
@aashish24 Done.

@doutriaux1 Until now, I had no idea about the json stuff. I'll take a look at it too.

**Comment by doutriaux1** [comment]
@sankhesh lots of failing tests... Also basic plot now fails:

```python
import os,sys,cdms2,vcs

f=cdms2.open(os.path.join(vcs.sample_data,"clt.nc"))
s=f("clt")
x=vcs.init()
x.plot(s)
```

```
python test.py
Traceback (most recent call last):
  File "test.py", line 6, in <module>
    x.plot(s)
  File "/lgm/uvcdat/2015-09-03/lib/python2.7/site-packages/vcs/Canvas.py", line 2524, in plot
    a = self.__plot(arglist, keyargs)
  File "/lgm/uvcdat/2015-09-03/lib/python2.7/site-packages/vcs/Canvas.py", line 3839, in __plot
    returned_kargs = self.backend.plot(*arglist, **keyargs)
  File "/lgm/uvcdat/2015-09-03/lib/python2.7/site-packages/vcs/VTKPlots.py", line 466, in plot
    vtk_backend_grid, vtk_backend_geo))
  File "/lgm/uvcdat/2015-09-03/lib/python2.7/site-packages/vcs/vcsvtk/pipeline2d.py", line 157, in plot
    self._plotInternal()
  File "/lgm/uvcdat/2015-09-03/lib/python2.7/site-packages/vcs/vcsvtk/boxfillpipeline.py", line 186, in _plotInternal
    for ind, mapper in enumerate(self._patternMappers):
TypeError: 'NoneType' object is not iterable
```

**Comment by sankhesh** [comment]
@doutriaux1 Could you please try with a new checkout? I had to force-push a couple of changes.

**Comment by aashish24** [comment]
@sankhesh looks like you still have 52 tests failing. Most of the has the same pattern.

**Comment by sankhesh** [comment]
@doutriaux1  All regressions pass now.

**Comment by doutriaux1** [comment]
Nice! thanks will play with it now and let you know. Will let our scientists know as well. @bonfils2 @durack1 I will let you know when it's ready.

**Comment by durack1** [comment]
@sankhesh are there some missing files in the test-suite? [On the macbot I see](http://aims4.llnl.gov:9981/master/macbot.llnl.gov:9982/UV-CDAT/uvcdat/b3cd32359e766e580d87fcd2ab96fd402ec2fc48/ctest__-j4__-D):
```
The following tests FAILED:
	302 - vcs_test_patterns (Failed)
	309 - vcs_test_isofill_patterns (Failed)
	310 - vcs_test_isofill_hatches (Failed)
	311 - vcs_test_hatches_patterns (Failed)
	460 - basemap_verify_import (Failed)
File "/Users/doutriaux1/uvcbot/build/uvcdat-testdata/baselines/vcs/test_vcs_hatches_patterns.png" not found.
File "/Users/doutriaux1/uvcbot/build/uvcdat-testdata/baselines/vcs/test_isofill_hatches.png" not found.
File "/Users/doutriaux1/uvcbot/build/uvcdat-testdata/baselines/vcs/test_isofill_patterns.png" not found.
```

**Comment by sankhesh** [comment]
@durack1 

Those files are sitting in a PR @ UV-CDAT/uvcdat-testdata#61 
Typically, the testdata should be updated everytime its built. No idea why the macbot fails to do so. The other machines seem to be fine.

**Comment by aashish24** [comment]
@sankhesh awesome dude!..

**Pull request merged**